src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml

```diff
@@ -565,6 +595,14 @@ struct
             :: !pub_input_gate_specs_rev
         done ;
 
+        (* finalize any pending generic constraint *)
+        ( match sys.pending_generic_gate with
+        | None ->
+            ()
+        | Some (l, r, o, coeffs) ->
+            add_row sys [| l; r; o |] Generic coeffs ;
```

Comment:
Doesn't this need to have 6 parameters (trailing 3 as zeros)?

Reply:
If you mean the variables (`[|l;r;o|]`), I don't think it makes sense to have zero variables. If you mean the coeffs, then it [gets padded with zeros on the rust side](https://github.com/o1-labs/proof-systems/pull/218/files#diff-26e568584dbb0ffa72d5ebde44a43a1f39d176c7c2f190aa933abbbc5a4e04aeR472).

There is an issue here though... the coeffs are not continuous:

<img width="681" alt="Screen Shot 2022-01-26 at 10 19 17 AM" src="https://user-images.githubusercontent.com/1316043/151222955-3033cc03-073d-4587-bebf-61864be14935.png">
 
so I need to either make them continuous in our layout, or do something here, so thanks for pointing that out :D

Reply:
although I'm still puzzled by your comment and wondering if I missed something..